app/models/email.rb

```diff
@@ -23,4 +23,11 @@ def self.timed_bulk_insert(records, batch_size)
 
     Metrics.email_bulk_insert(batch_size) { insert_all!(records) }
   end
+
+  def mark_as_spam!
+    unless marked_as_spam?
+      update!(marked_as_spam: true)
```

Comment:
Do we gain value storing this on an email model? I guess we don't expect it to be used for anything and deleted within a few days. 

It bugs me a bit as, aside from this, the model shouldn't be amended once we've finished sending and could be treated as immutable.

Reply:
Moving this to the service 👌

Reply:
@kevindew Did you mean the `marked_as_spam` attribute itself?

Reply:
Yup, the attribute itself.

Reply:
Indeed we don't do much with that attribute. We now use it only to check whether a given email has already been marked as spam(https://github.com/alphagov/email-alert-api/pull/1336/files#diff-7bc42ad20bdcd55a897f3905ef82e794R12). Do you think it's ok if we remove that check @kevindew ?

Reply:
Sorry for wading in: it does have a slight value, because it de-dupes a user marking the same email as spam multiple times. Otherwise, we could skip this metric and just use the request logs from Kibana.*

*when I looked at the request logs for the spam endpoint, the data seemed a bit off to me. Perhaps it's worth another look, as it's still kind of relevant if a user marks the same email as spam multiple times.

Reply:
Sure, that makes sense. Bummer for the immutable aspect of the model mind. 

Do we know how it works at AWS SES/Notify if the same email gets marked multiple times? Might be worth checking that Notify don't retry us each time we return a 404.

Reply:
Regarding those 404s, it does indeed look like it's caused by users unsubscribing after the email being marked as spam has been archived. The way around this would be for `/spam-reports` endpoint to use a unique identifier for the subscriber (e.g. subscriber ID, or email address) and use that to fetch the subscriber record and unsubscribe it from everything.

Reply:
Are you confident this happens more than the subscriber_id being nil? 

The bit that intrigues me is whether Notify matches the complaint to their data or not. I think they also have a 7 day retention period of emails so if they're matching the email to their database in theory our database should be pretty much the same. Whereas they could be sending them through whether they have the email or not which could mean it comes from any time.